Onitu Client

entwanne opened the pull request.

entwanne committed:
Majorodomo for managing remote drivers (clients)

entwanne committed:
Adding curve authentication in Majordomo

entwanne committed:
Add handlers and metadata's serializer

entwanne committed:
Requests from remote driver

entwanne committed:
Fix race-condition at client's startup

entwanne committed:
Delete comment

entwanne committed:
Handle move and delete

entwanne committed:
Add catch for ContextTerminated in remote_driver

entwanne committed:
Properly close sockets in remote_drivert

entwanne committed:
Majordomo cleanup

entwanne committed:
Clean Majordomo

entwanne committed:
Clean code for remote_driver

entwanne committed:
Handle DriverError and ServiceError

entwanne committed:
Add lock in remote_driver handlers

entwanne committed:
Disconnect remote drivers

entwanne committed:
Adapt remote-driver to new architecture

entwanne committed:
New broker

entwanne committed:
Add parameters for Majordomo configuration

entwanne committed:
Flake8 remote-driver

entwanne committed:
Remote driver entry_db

entwanne committed:
Adapt remote-driver to unicode

entwanne committed:
Clean msgpack

entwanne committed:
New rules majordomo

entwanne committed:
Add folders in remote driver

entwanne committed:
Add operation to reset referee rules

entwanne committed:
Receive configuration from client

entwanne committed:
Merge client in main repository

entwanne committed:
Fix build

entwanne committed:
Can disable Majordomo auth

entwanne committed:
Client fault-tolerance

k4nar commented on client/__main__.py:
```diff
@@ -0,0 +1,123 @@
+logger = None
+
+
+def get_logs_dispatcher(uri=None, debug=False):
```
should be in utils

k4nar commented on client/__main__.py:
```diff
@@ -0,0 +1,123 @@
+    return uri, subscriber.dispatch_in_background(setup=NestedSetup(handlers))
+
+
+def get_setup(setup_file):
```
should also be in utils

k4nar replied:
Also the version from `onitu.__main__` should be the one moved to the utils, as it has changed.

k4nar commented on onitu/plug/client/__init__.py:
```diff
@@ -0,0 +1,4 @@
+from .plug import PlugProxy as Plug
+from .exceptions import DriverError, ServiceError
+
+__all__ = Plug, DriverError, ServiceError
```
Should be a list of strings.

k4nar commented on drivers/remote_driver/onitu_remote_driver/remote_driver.py:
```diff
@@ -0,0 +1,66 @@
+    global remote_socket, handlers_socket
+
+    print("Launching driver")
+    print(plug.options)
```
please remove prints

k4nar commented on onitu/majordomo/heartbeat.py:
```diff
@@ -0,0 +1,23 @@
+
+@HeartBeatBroker.handle('heartbeat')
+def _heartbeat(broker, relay, msg):
+    print('ping', msg.src_id)
```
remove print

k4nar commented on onitu/plug/client/heartbeat.py:
```diff
@@ -0,0 +1,34 @@
+                    self.callback()
+                    return
+                msg = self.socket.recv()
+                print(msg)
```
remove print

k4nar commented on onitu/referee/referee.py:
```diff
@@ -152,8 +153,17 @@ def _handle_update(self, fid, source):
+        self.folders = Folder.get_folders(
+            self.escalator, self.services, self.logger
+        )
+        print(self.services)
```
print, again :)

k4nar commented on onitu/__main__.py:
```diff
@@ -205,6 +216,34 @@ def main():
+    parser.add_argument(
+        '--majordomo-keys-dir',
+        help="Directory where clients' public keys are stored",
+        default='keys'
```
Should be prefixed by the config dir, like the setup for instance.

k4nar commented:
It would be really great to have a README with some explanation. We'll add it to the docs later.

wapiflapi committed:
Address most (all) of k4nar's comments on PR #138.

I'm not in a position to test the client right now, normaly this
shouldn't break anything. (onitu core & driver tests are OK). But
someone should check if the client still starts before merging this :)

entwanne committed:
Fix client

entwanne committed:
Heartbeat addr

entwanne committed:
Fix move_file request in the ProxyPlug :)

entwanne committed:
Logging client
